spec/pg/connection_spec.cr

```diff
@@ -104,3 +104,23 @@ describe PG, "#read_next_row_start" do
     end
   end
 end
+
+describe PG, "#time_zone" do
+  it "reads time zones from server parameters" do
+    with_connection do |db|
+      db.exec "SET TIME ZONE 'America/Los_Angeles'"
+      now = db.query_one "SELECT now()", as: Time
+
+      now.location.should eq Time::Location.load("America/Los_Angeles")
+
+      db.transaction do |txn|
+        db.exec "SET LOCAL TIME ZONE 'America/New_York'"
+        now = db.query_one "SELECT now()", as: Time
+        now.location.should eq Time::Location.load("America/New_York")
+      end
+
+      now = db.query_one "SELECT now()", as: Time
+      now.location.should eq Time::Location.load("America/Los_Angeles")
+    end
+  end
```

Comment:
TIL server parameters (on the wire protocol, that is), so we don't have to store the time zone separately, and it will automatically be pushed to the client when `SET TIME ZONE` or `SET LOCAL TIME ZONE` is called. `PQ::Connection` already [stores all the server parameters as they come through](https://github.com/will/crystal-pg/blob/cafe7220c286fc5413b7aeebbec3d4c9efd06e62/src/pq/connection.cr#L223) so all we have to do is use it.